tests/components/threshold/test_binary_sensor.py

```diff
@@ -1,384 +1,387 @@
 
-        assert state.state == "on"
+    assert state.state == "on"
 
-        self.hass.states.set("sensor.test_monitored", 13)
-        self.hass.block_till_done()
+    hass.states.async_set("sensor.test_monitored", 13)
+    await hass.async_block_till_done()
 
-        state = self.hass.states.get("binary_sensor.threshold")
+    state = hass.states.get("binary_sensor.threshold")
 
-        assert state.state == "on"
+    assert state.state == "on"
 
-        self.hass.states.set("sensor.test_monitored", 12)
-        self.hass.block_till_done()
+    hass.states.async_set("sensor.test_monitored", 12)
+    await hass.async_block_till_done()
 
-        state = self.hass.states.get("binary_sensor.threshold")
+    state = hass.states.get("binary_sensor.threshold")
 
-        assert state.state == "off"
+    assert state.state == "off"
 
-        self.hass.states.set("sensor.test_monitored", 17)
-        self.hass.block_till_done()
+    hass.states.async_set("sensor.test_monitored", 17)
+    await hass.async_block_till_done()
 
-        state = self.hass.states.get("binary_sensor.threshold")
+    state = hass.states.get("binary_sensor.threshold")
 
-        assert state.state == "off"
+    assert state.state == "off"
 
-        self.hass.states.set("sensor.test_monitored", 18)
-        self.hass.block_till_done()
+    hass.states.async_set("sensor.test_monitored", 18)
+    await hass.async_block_till_done()
 
-        state = self.hass.states.get("binary_sensor.threshold")
+    state = hass.states.get("binary_sensor.threshold")
 
-        assert state.state == "on"
+    assert state.state == "on"
 
-    def test_sensor_in_range_no_hysteresis(self):
-        """Test if source is within the range."""
-        config = {
-            "binary_sensor": {
-                "platform": "threshold",
-                "lower": "10",
-                "upper": "20",
-                "entity_id": "sensor.test_monitored",
-            }
+
+async def test_sensor_in_range_no_hysteresis(hass):
+    """Test if source is within the range."""
+    config = {
+        "binary_sensor": {
+            "platform": "threshold",
+            "lower": "10",
+            "upper": "20",
+            "entity_id": "sensor.test_monitored",
         }
+    }
+
+    assert await async_setup_component(hass, "binary_sensor", config)
+    await hass.async_block_till_done()
 
-        assert setup_component(self.hass, "binary_sensor", config)
+    hass.states.async_set(
+        "sensor.test_monitored", 16, {ATTR_UNIT_OF_MEASUREMENT: TEMP_CELSIUS}
+    )
+    await hass.async_block_till_done()
 
-        self.hass.states.set(
-            "sensor.test_monitored", 16, {ATTR_UNIT_OF_MEASUREMENT: TEMP_CELSIUS}
-        )
-        self.hass.block_till_done()
+    state = hass.states.get("binary_sensor.threshold")
 
-        state = self.hass.states.get("binary_sensor.threshold")
+    assert "sensor.test_monitored" == state.attributes.get("entity_id")
```

Comment:
We should put the expected value on the right side of the `==`. 
```suggestion
    assert state.attributes.get("entity_id") == "sensor.test_monitored"
```